components/builder-depot/src/server.rs

```diff
@@ -1236,6 +1244,15 @@ fn download_package(req: &mut Request) -> IronResult<Response> {
 
     match route_message::<OriginPackageGet, OriginPackage>(req, &ident_req) {
         Ok(package) => {
+            let temp_path = depot.archive_path(package.get_ident(), &target);
+            let temp_ident = ident.to_owned().into();
+            match s3handler.download(&temp_path, &temp_ident, &target) {
+                Ok(_target) => {
+                    Response::with(status::Ok);
+                    _target
+                }
+                Err(_) => return Ok(Response::with(status::NotFound)),
+            }
```

Comment:
I'm not sure I understand this logic.  We are downloading into a temp path, but we're not renaming it into an archive path.  How is it getting translated into the archive path?  We should either just completely get rid of the archive_path function (which I think is the right thing to do - because on prod we're going to serve mostly from fastly, and on-prem we'll serve from a local store), or if we really really want to use an on-disk cache, then it should be a much simpler path than constructing an elaborate archive path.

Reply:
This has been simplified. The archive path is actually just the hartfile + /pkgs. We're no longer generating a complex path we simply use /pkgs as the caching location. A follow-on PR could definitely be made to switch all of this pathing over to using a tempdir or tempfile abstraction but doing so would be a more significat refactor than originally thought. For now hopefully the simplification is a reasonable enough middle-ground.

Reply:
Ok, cool!